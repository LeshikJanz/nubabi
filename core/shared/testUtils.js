/* eslint-disable import/no-extraneous-dependencies */
import configureMockStore from 'redux-mock-store';
import { createEpicMiddleware } from 'redux-observable';
import firebaseMock from 'firebase-mock';

export const setupEpic = epic => {
  const epicMiddleware = createEpicMiddleware(epic);
  const mockStore = configureMockStore([epicMiddleware]);

  return {
    middleware: epicMiddleware,
    mockStore,
    afterEach: () => {
      epicMiddleware.replaceEpic(epic);
    },
  };
};

export const mockFirebase = () => {
  const mockDb = new firebaseMock.MockFirebase();
  const mockAuth = new firebaseMock.MockFirebase();
  const mockSdk = firebaseMock.MockFirebaseSdk(
    path => {
      return path ? mockDb.child(path) : mockDb;
    },
    () => {
      return mockAuth;
    },
  );

  return mockSdk;
};
