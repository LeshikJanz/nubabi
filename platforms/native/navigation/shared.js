import React from 'react';
import { pathOr } from 'ramda';
import LeftHeader from './LeftHeader';
import RightHeader from './RightHeader';

const babyNameTitle = ({ state }) =>
  pathOr('Nubabi', ['params', 'babyName'], state);

export function getBabyTitle(navigation) {
  if (!navigation) {
    throw new Error('an screen hasnt been updated, check stack');
  }

  return babyNameTitle(navigation);
}

export const getTabHeaders = navigate => ({
  left: <LeftHeader navigate={navigate} />,
  right: <RightHeader navigate={navigate} />,
});
