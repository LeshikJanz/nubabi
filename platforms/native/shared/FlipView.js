// @flow
import React, { Component } from 'react';
import { View, Animated } from 'react-native';

const styles = {
  animatedContainer: {
    flex: 1,
  },
};

type Props = {
  renderFront: Function,
  renderBack: Function,
  velocity: number,
  tension: number,
  friction: number,
};

export class FlipCard extends Component {
  props: Props;

  state = {
    animatedValue: new Animated.Value(0),
    isFlipped: true,
  };

  componentDidUpdate(prevProps: Props, prevState: *) {
    if (this.state.isFlipped !== prevState.isFlipped) {
      this._flippedCard();
    }
  }

  _flipToggleCard = () => {
    this.setState({ isFlipped: !this.state.isFlipped });
  };

  _flippedCard() {
    Animated.spring(this.state.animatedValue, {
      toValue: 0, // Returns to the start
      velocity: this.props.velocity, // Velocity makes it move
      tension: this.props.tension, // Slow
      friction: this.props.friction, // Oscillate a lot
    }).start();
  }

  flippedCardView(isFlipped) {
    if (isFlipped) {
      return this.props.renderFront(this._flipToggleCard);
    }
    return this.props.renderBack(this._flipToggleCard);
  }

  render() {
    const perspective = 2000;
    const rotateY = this.state.animatedValue.interpolate({
      inputRange: [0, 1],
      outputRange: ['0deg', '180deg'],
    });
    return (
      <View style={styles.animatedContainer}>
        <Animated.View
          style={[
            styles.animatedContainer,
            { transform: [{ rotateY }, { perspective }] },
          ]}
        >
          {this.flippedCardView(this.state.isFlipped)}
        </Animated.View>
      </View>
    );
  }
}

export default FlipCard;
