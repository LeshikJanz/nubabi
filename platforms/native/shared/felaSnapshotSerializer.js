// WIP
// prettier-ignore
const felaSnapshotSerializer = {
  test(val) {
    return val && val.felaRules && val.$$typeof === Symbol.for('react.test.json');
  },

  print(val, print) {
    const styles = val.felaRules;
    val.felaRules = false;

    return `style = Object ${JSON.stringify(styles, null, 4)}\n\n${print(val)}`;
  },
};

module.exports = felaSnapshotSerializer;
