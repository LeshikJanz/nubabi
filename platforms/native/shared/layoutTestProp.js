export default {
  fontScale: 2,
  height: 1334,
  scale: 2,
  width: 750,
  viewportWidth: 750,
  viewportHeight: 1334,
};
