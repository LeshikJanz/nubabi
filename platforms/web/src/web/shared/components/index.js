import About from "./About";
import AsyncComponent from "./AsyncComponent";
import AuthenticatedRoute from "./AuthenticatedRoute";
import Header from "./Header";
import Home from "./Home";
import Loader from "./Loader";
import Test from "./Test";

export default {
  About,
  AsyncComponent,
  AuthenticatedRoute,
  Header,
  Home,
  Loader,
  Test
};
