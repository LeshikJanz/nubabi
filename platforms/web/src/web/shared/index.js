import * as actions from "./actions";
export { default as containers } from "./containers";
export { default as reducer } from "./reducer";
export { default as epics } from "./epics";
export { actions };
