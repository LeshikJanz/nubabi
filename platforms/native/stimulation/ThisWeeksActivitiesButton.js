// @flow
import type { LayoutProps } from 'core/types';
import React from 'react';
import { ImageBackground } from 'react-native';
import moment from 'moment';
import { Box, Card, Icon, Overlay, Text, withLayout } from '../components';
import theme from 'core/themes/defaultTheme';
import formatDate from 'core/helpers/formatDate';

const background = require('core/images/gross_motor_large.jpg');

type Props = {
  onPress: () => void,
  layout: LayoutProps,
};

export const ThisWeeksActivitiesButton = ({ onPress, layout }: Props) => {
  const title = "This Week's Stimulation Guide";

  const dimensions = {
    width: layout.viewportWidth - 20,
    height: layout.viewportWidth * 0.6,
  };

  const date = moment()
    .startOf('week')
    .toDate();

  return (
    <Box contentSpacing>
      <Card padding={0} onPress={onPress}>
        <Box style={() => ({ borderRadius: 4, overflow: 'hidden' })}>
          <ImageBackground
            source={background}
            style={[{ flex: 1 }, dimensions]}
            resizeMode="cover"
          >
            <Overlay
              style={{
                justifyContent: 'flex-start',
                alignItems: 'flex-start',
                paddingTop: 16,
                paddingLeft: 16,
              }}
            >
              <Box
                alignItems="center"
                justifyContent="center"
                flexDirection="row"
              >
                <Icon
                  name="md-calendar"
                  size={20}
                  color="white"
                  style={{ marginTop: 2 }}
                />
                <Text color="white" marginLeft={0.5} bold>
                  {formatDate(date)}
                </Text>
              </Box>
            </Overlay>
          </ImageBackground>
          <Box contentSpacing flexDirection="row" justifyContent="center">
            <Text flex={1} size={6}>
              {title}
            </Text>
            <Box alignItems="center" justifyContent="center">
              <Icon
                color={theme.colors.secondary}
                name="ios-arrow-forward"
                size={20}
                style={{ marginTop: 2 }}
              />
            </Box>
          </Box>
        </Box>
      </Card>
    </Box>
  );
};

export default withLayout(ThisWeeksActivitiesButton);
