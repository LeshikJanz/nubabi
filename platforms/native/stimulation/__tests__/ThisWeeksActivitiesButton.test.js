import 'react-native';
import React from 'react';
import { ThisWeeksActivitiesButton } from '../ThisWeeksActivitiesButton';
import { expectRender, layoutTestProp } from '../../shared/testUtils';

Date.now = jest.fn(() => new Date(2017, 7, 22));

test('it renders correctly', () => {
  expectRender(<ThisWeeksActivitiesButton layout={layoutTestProp} />);
});
