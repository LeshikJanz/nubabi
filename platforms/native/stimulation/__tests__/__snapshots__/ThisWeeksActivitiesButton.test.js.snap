// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
style = Object {
    "paddingBottom": 10,
    "paddingLeft": 10,
    "paddingRight": 10,
    "paddingTop": 10,
    "borderColor": "#9EABBC",
    "borderBottomWidth": 0,
    "borderLeftWidth": 0,
    "borderRightWidth": 0,
    "borderTopWidth": 0,
    "marginBottom": 16,
    "flex": 1,
    "backgroundColor": "#FFFFFF",
    "alignItems": "center",
    "borderBottomLeftRadius": 4,
    "borderBottomRightRadius": 4,
    "borderTopLeftRadius": 4,
    "borderTopRightRadius": 4,
    "justifyContent": "center",
    "shadowColor": "#000",
    "shadowOpacity": 0.15,
    "shadowRadius": 3,
    "shadowOffset": {
        "height": 1,
        "width": 0
    },
    "borderRadius": 4,
    "overflow": "hidden",
    "flexDirection": "row",
    "marginLeft": 8,
    "fontSize": 19.147659412894374,
    "lineHeight": 32,
    "color": "#454D56",
    "fontFamily": "SF Pro Display",
    "fontWeight": "700"
}

<View
  contentSpacing={true}
  style={
    Object {
      "borderBottomWidth": 0,
      "borderColor": "#9EABBC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "paddingBottom": 10,
      "paddingLeft": 10,
      "paddingRight": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    onLayout={[Function]}
    onPress={undefined}
    style={
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#FFFFFF",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderBottomWidth": 0,
        "borderColor": "#9EABBC",
        "borderLeftWidth": 0,
        "borderRightWidth": 0,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "borderTopWidth": 0,
        "flex": 1,
        "justifyContent": "center",
        "marginBottom": 16,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.15,
        "shadowRadius": 3,
      }
    }
  >
    <View
      style={
        Object {
          "borderBottomWidth": 0,
          "borderColor": "#9EABBC",
          "borderLeftWidth": 0,
          "borderRadius": 4,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "height": 450,
              "width": 730,
            },
          ]
        }
      >
        <Image
          resizeMode="cover"
          source={
            Object {
              "testUri": "../../../core/images/gross_motor_large.jpg",
            }
          }
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(116, 130, 148)",
                  "bottom": 0,
                  "flex": 1,
                  "justifyContent": "center",
                  "left": 0,
                  "opacity": 0.4,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
                Object {},
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "transparent",
                  "flex": 1,
                  "flexDirection": "column",
                  "justifyContent": "center",
                },
                Object {
                  "alignItems": "flex-start",
                  "justifyContent": "flex-start",
                  "paddingLeft": 16,
                  "paddingTop": 16,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "borderBottomWidth": 0,
                  "borderColor": "#9EABBC",
                  "borderLeftWidth": 0,
                  "borderRightWidth": 0,
                  "borderTopWidth": 0,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={false}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 20,
                    },
                    Object {
                      "marginTop": 2,
                    },
                    Object {
                      "fontFamily": "Ionicons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                  ]
                }
              >
                
              </Text>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "borderBottomWidth": 0,
                    "borderColor": "#9EABBC",
                    "borderLeftWidth": 0,
                    "borderRightWidth": 0,
                    "borderTopWidth": 0,
                    "color": "#FFFFFF",
                    "fontFamily": "SF Pro Text",
                    "fontSize": 13,
                    "fontWeight": "700",
                    "lineHeight": 16,
                    "marginLeft": 8,
                  }
                }
              >
                AUGUST 20 2017
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        contentSpacing={true}
        style={
          Object {
            "borderBottomWidth": 0,
            "borderColor": "#9EABBC",
            "borderLeftWidth": 0,
            "borderRightWidth": 0,
            "borderTopWidth": 0,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "borderBottomWidth": 0,
              "borderColor": "#9EABBC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "color": "#454D56",
              "flex": 1,
              "fontFamily": "SF Pro Display",
              "fontSize": 19.147659412894374,
              "lineHeight": 32,
            }
          }
        >
          This Week's Stimulation Guide
        </Text>
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomWidth": 0,
              "borderColor": "#9EABBC",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#748294",
                  "fontSize": 20,
                },
                Object {
                  "marginTop": 2,
                },
                Object {
                  "fontFamily": "Ionicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
